reporter/src/main/kotlin/reporters/OpossumReporter.kt

```diff
@@ -0,0 +1,599 @@
+                    "packageVersion" to id?.version,
+
+                    "copyright" to copyright,
+                    "licenseName" to license?.toString(),
+
+                    "url" to url,
+
+                    "preSelected" to preselected,
+                    "followUp" to "FOLLOW_UP".takeIf { followUp },
+                    "excludeFromNotice" to excludeFromNotice,
+
+                    "comment" to comment
+                )
+            )
+        }
+
+        fun matches(other: OpossumSignal): Boolean =
+            source == other.source
+                    && id == other.id
+                    && url == other.url
+                    && license == other.license
+                    && copyright == other.copyright
+                    && comment == other.comment
+                    && preselected == other.preselected
+    }
+
+    data class OpossumResources(
+        val tree: MutableMap<String, OpossumResources> = mutableMapOf()
+    ) {
+        fun addResource(pathPieces: List<String>) {
+            if (pathPieces.isEmpty()) { return }
+
+            val head = pathPieces.first()
+            val tail = pathPieces.drop(1)
+
+            if (head !in tree) { tree[head] = OpossumResources() }
+            tree.getValue(head).addResource(tail)
+        }
+
+        fun addResource(path: String) {
+            val pathPieces = path.split("/")
+                .filter { it.isNotEmpty() }
+            addResource(pathPieces)
+        }
+
+        fun isFile() = tree.isEmpty()
+
+        fun isPathAFile(path: String): Boolean {
+            val pathPieces = path.split("/")
+                .filter { it.isNotEmpty() }
+            return isPathAFile(pathPieces)
+        }
+
+        private fun isPathAFile(pathPieces: List<String>): Boolean {
+            if (pathPieces.isEmpty()) { return isFile() }
+
+            val head = pathPieces.first()
+            val tail = pathPieces.drop(1)
+
+            if (head !in tree) { return true }
+            return tree.getValue(head).isPathAFile(tail)
+        }
+
+        fun toJson(): Map<*, *> = tree.mapValues { (_, v) -> if (v.isFile()) 1 else v.toJson() }
+
+        fun toFileList(): Set<String> =
+            tree.entries.flatMapTo(mutableSetOf()) { (key, value) ->
+                value.toFileList().map { resolvePath(key, it, isDirectory = false) }
+            }.plus("/")
+    }
+
+    data class OpossumFrequentLicense(
+        var shortName: String,
+        var fullName: String?,
+        var defaultText: String?
+    ) : Comparable<OpossumFrequentLicense> {
+        fun toJson(): Map<*, *> = sortedMapOf(
+            "shortName" to shortName,
+            "fullName" to fullName,
+            "defaultText" to defaultText
+        )
+
+        override fun compareTo(other: OpossumFrequentLicense) =
+            compareValuesBy(this, other,
+            { it.shortName },
+            { it.fullName },
+            { it.defaultText })
```

Comment:
Formatting:

            compareValuesBy(this, other,
                { it.shortName },
                { it.fullName },
                { it.defaultText }
            )